packages/core/src/browser/preferences/preference-service.ts

```diff
@@ -16,27 +16,67 @@
 
 // tslint:disable:no-any
 
-import { JSONExt } from '@phosphor/coreutils';
 import { injectable, inject, postConstruct } from 'inversify';
 import { FrontendApplicationContribution } from '../../browser';
 import { Event, Emitter, DisposableCollection, Disposable, deepFreeze } from '../../common';
 import { Deferred } from '../../common/promise-util';
 import { PreferenceProvider } from './preference-provider';
 import { PreferenceSchemaProvider } from './preference-contribution';
+import URI from '../../common/uri';
 
 export enum PreferenceScope {
+    Default,
     User,
-    Workspace
+    Workspace,
+    Folder
 }
 
-export interface PreferenceChangedEvent {
-    changes: PreferenceChange[]
+export namespace PreferenceScope {
+    export function is(scope: any): scope is PreferenceScope {
+        return typeof scope === 'number' && getScopes().findIndex(s => s === scope) >= 0;
+    }
+
+    export function getScopes(): PreferenceScope[] {
+        return Object.keys(PreferenceScope)
+            .filter(k => typeof PreferenceScope[k as any] === 'string')
+            .map(v => <PreferenceScope>Number(v));
+    }
+
+    export function getReversedScopes(): PreferenceScope[] {
+        return getScopes().reverse();
+    }
+
+    export function getScopeNames(scope?: PreferenceScope): string[] {
+        const names: string[] = [];
+        const allNames = Object.keys(PreferenceScope)
+            .filter(k => typeof PreferenceScope[k as any] === 'number');
+        if (scope) {
+            for (const name of allNames) {
+                if ((<any>PreferenceScope)[name] <= scope) {
+                    names.push(name);
+                }
+            }
+        }
+        return names;
+    }
+
+    export function fromString(strScope: string): PreferenceScope | undefined {
+        switch (strScope) {
+            case 'application':
+                return PreferenceScope.User;
+            case 'window':
+                return PreferenceScope.Workspace;
+            case 'resource':
+                return PreferenceScope.Folder;
+        }
+    }
 }
 
 export interface PreferenceChange {
     readonly preferenceName: string;
     readonly newValue?: any;
     readonly oldValue?: any;
+    readonly scope: PreferenceScope;
```

Comment:
How can i check whether a change affected me in some context?

I remember you had something like `affects(resourceUri?: string): boolean`.

I don't understand how i can use  `scope`  to check affection. Maybe  `scope` also should not be on an event?

Reply:
removed `scope` and added `affects()` back. Thank you for the review !